### PR TITLE
Reducing tsbuildinfo size further

### PR DESCRIPTION
**Pull request by sheetalkamat**
Here are the numbers with one of the internal tsbuildinfo
4de704d Only emit compiler options that affect
```
04/15/2021  01:26 PM           670,576 infoNew.json
04/15/2021  01:26 PM         1,023,762 infoOriginal.json
```
a422f9e: Dont emit affectsGlobalScope if its not true

```
04/15/2021  01:38 PM           612,256 infoNew.json
04/15/2021  01:38 PM         1,023,762 infoOriginal.json
```
658587d: FileInfo is encoded as string if FileInfo.version === FileInfo.signature && !FileInfo.affectsGlobalScope, otherwise property signature is encoded as false if it was undefined, undefined if FileInfo.version === FileInfo.signature and actual signature otherwise
```
04/15/2021  03:47 PM           410,798 infoNew.json
04/15/2021  03:47 PM         1,023,762 infoOriginal.json
```

### CODE REVIEW

File: src/compiler/builder.ts [hunk 1]
```diff
@@ -730,12 +743,21 @@ namespace ts {
...
         let fileIdsList: (readonly number[])[] | undefined;
         let fileNamesToFileIdListId: ESMap<string, number> | undefined;
-        const fileInfos = arrayFrom(state.fileInfos.entries(), ([key, value]) => {
+        const fileInfos = arrayFrom(state.fileInfos.entries(), ([key, value]): ProgramBuildInfoFileInfo => {
             // Ensure fileId
             const fileId = toFileId(key);
             Debug.assert(fileNames[fileId - 1] === relativeToBuildInfo(key));
             const signature = state.currentAffectedFilesSignatures && state.currentAffectedFilesSignatures.get(key);
-            return signature === undefined ? value : { version: value.version, signature, affectsGlobalScope: value.affectsGlobalScope };
+            const actualSignature = signature === undefined ? value.signature : signature;
```
Comment (amcasey) [inline src/compiler/builder.ts:751]: Is this `signature ?? value.signature`?

**Review by amcasey — APPROVED** [review]
LGTM.  Just measuring against the motivating code base (ETA one hour).

File: src/compiler/builder.ts (hunk 1 above)
Comment (sheetalkamat) [reply src/compiler/builder.ts:751]: yes sorry i kind of copied from original code so didnt clean it up later..

**Comment by amcasey** [comment]
tsbuildinfo size dropped from 1345MB to 467MB and total build time dropped ~15%. 🎉

File: src/compiler/builder.ts (hunk 1 above)
Comment (sheetalkamat) [inline src/compiler/builder.ts:751]:
```suggestion
            const actualSignature = signature ?? value.signature;
```

**Pull request merged**